app/Tools/LumenMailer.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * Ushahidi Mailer
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Tools;
+
+use Ushahidi\Core\Tool\Mailer as MailerContract;
+use Illuminate\Contracts\Mail\Mailer;
+use Illuminate\Support\Str;
+
+class LumenMailer implements MailerContract
+{
+    public function __construct(Mailer $mailer, $siteConfig, $clientUrl)
+    {
+        $this->mailer = $mailer;
+        $this->siteConfig = $siteConfig;
+        $this->clientUrl = $clientUrl;
+    }
+
+    public function send($to, $type, Array $params = null)
+    {
+        $method = "send" . Str::ucfirst($type);
+        if (method_exists($this, $method)) {
+            $this->$method($to, $params);
+        } else {
+            // Exception
+            throw new Exception('Unsupported mail type: ' + $type);
+        }
+    }
+
+    protected function sendResetpassword($to, $params)
```

Comment:
I'm not sure yet. I suspect we might want to replace it entirely with something much closer to Laravel's Mailables